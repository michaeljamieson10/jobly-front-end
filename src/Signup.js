import React, { useContext } from 'react';
import SignupForm from './SignupForm';
import JoblyApi from './JoblyApi';
import UserContext from "./UserContext";

function Signup() {
  const { setCurrentUser } = useContext(UserContext);
  async function createUser(data) {
    let user = await JoblyApi.createNewUser(data);
    setCurrentUser(user);
  }
  return (
  <>
  <SignupForm createUser={createUser}/>
  </>
  );
}

export default Signup;