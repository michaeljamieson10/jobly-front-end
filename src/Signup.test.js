import React from "react";
import { render } from "@testing-library/react";
import Signup from "./Signup";
import { MemoryRouter } from "react-router";
import { UserProvider } from "./testUtils";

it("renders without crashing", function() {
  render(
    <UserProvider>
      <Signup />
    </UserProvider>
  );
});

it("matches snapshot", function() {
  const { asFragment } = render(
    <UserProvider>
      <Signup />
    </UserProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
