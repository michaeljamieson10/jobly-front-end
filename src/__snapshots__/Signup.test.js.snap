// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <section
      class="col-md-10"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <form>
            <p>
              <label
                for="username"
              >
                Username:
              </label>
              <input
                id="username"
                name="username"
                value=""
              />
            </p>
            <p>
              <label
                for="first_name"
              >
                First Name:
              </label>
              <input
                id="first_name"
                name="first_name"
                value=""
              />
            </p>
            <p>
              <label
                for="last_name"
              >
                Last Name:
              </label>
              <input
                id="last_name"
                name="last_name"
                value=""
              />
            </p>
            <p>
              <label
                for="email"
              >
                Email:
              </label>
              <input
                id="email"
                name="email"
                value=""
              />
            </p>
            <p>
              <label
                for="password"
              >
                Password:
              </label>
              <input
                id="password"
                name="password"
                type="password"
                value=""
              />
            </p>
            <button
              class="btn btn-primary"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
